src/panels/lovelace/cards/hui-sensor-graph-card.js

```diff
@@ -0,0 +1,273 @@
+import { LitElement, html } from '@polymer/lit-element';
+
+import '../../../components/ha-card.js';
+import '../../../components/ha-icon.js';
+
+import computeStateName from '../../../common/entity/compute_state_name.js';
+import stateIcon from '../../../common/entity/state_icon.js';
+
+import EventsMixin from '../../../mixins/events-mixin.js';
+
+class HuiSensorGraphCard extends EventsMixin(LitElement) {
+  set hass(hass) {
+    this._hass = hass;
+    const entity = hass.states[this.config.entity];
+    if (entity && this.entity !== entity) {
+      this.entity = entity;
+      this._getHistory();
+    }
+  }
+
+  static get properties() {
+    return {
+      _hass: Object,
+      config: Object,
+      entity: Object,
+      line: String
+    };
+  }
+
+  setConfig(config) {
+    if (!config.entity || config.entity.split('.')[0] !== 'sensor') {
+      throw new Error('Specify an entity from within the sensor domain.');
+    }
+
+    config.icon = config.icon || false;
```

Comment:
We should not change the incoming data. Instead we should create a new object that includes our defaults.

```
const cardConfig = Object.assign({
  hours_to_show: 24,
  // other defaults
}, config);
cardConfig.accuracy = Number(cardConfig.accuracy);
```